src/app/calculator/motors/motor-performance/motor-performance-chart/motor-performance-chart.component.ts

```diff
@@ -0,0 +1,321 @@
+import { Component, OnInit, HostListener, ViewChild, ElementRef, Input, ChangeDetectorRef, SimpleChanges } from '@angular/core';
+import { Settings } from '../../../../shared/models/settings';
+import { DataPoint, SimpleChart, TraceCoordinates } from '../../../../shared/models/plotting';
+import { FormGroup } from '@angular/forms';
+import { graphColors } from '../../../../phast/phast-report/report-graphs/graphColors';
+
+import * as Plotly from 'plotly.js';
+import { MotorPerformanceChartService, MotorPoint } from '../motor-performance-chart.service';
+
+@Component({
+  selector: 'app-motor-performance-chart',
+  templateUrl: './motor-performance-chart.component.html',
+  styleUrls: ['./motor-performance-chart.component.css']
+})
+export class MotorPerformanceChartComponent implements OnInit {
+  @Input()
+  performanceForm: FormGroup;
+  @Input()
+  toggleCalculate: boolean;
+  @Input()
+  settings: Settings;
+
+  @HostListener('document:keyup', ['$event'])
+  closeExpandedGraph(event) {
+    if (this.expanded) {
+      if (event.code === 'Escape') {
+        this.contractChart();
+      }
+    }
+  }
+
+  // DOM
+  @ViewChild("ngChartContainer", { static: false }) ngChartContainer: ElementRef;
+  @ViewChild('dataSummaryTable', { static: false }) dataSummaryTable: ElementRef;
+  dataSummaryTableString: any;
+  tabPanelChartId: string = 'tabPanelDiv';
+  expandedChartId: string = 'expandedChartDiv';
+  currentChartId: string = 'tabPanelDiv';
+
+  // Tooltips
+  hoverBtnGridLines: boolean = false;
+  displayGridLinesTooltip: boolean = false;
+  hoverBtnExpand: boolean = false;
+  displayExpandTooltip: boolean = false;
+  hoverBtnCollapse: boolean = false;
+  displayCollapseTooltip: boolean = false;
+  expanded: boolean = false;
+
+  // Graphing
+  selectedDataPoints: Array<MotorPoint>;
+  performanceChart: SimpleChart;
+  traceNames = {
+    0: 'Current (FLA)',
+    1: 'Power',
+    2: 'Efficiency'
+  };
+
+  curveChanged: boolean = false;
+  graphColors: Array<string>;
+
+  tempMotorPower: number;
+  tempRpm: number;
+  tempEfficiencyClass: string;
+  tempVoltage: number;
+  tempAmps: number;
+  tempLineFrequency: string;
+  motorPointColors: Array<string>;
+
+  firstChange: boolean = true;
+  constructor(private performanceChartService: MotorPerformanceChartService, private cd: ChangeDetectorRef) { }
+
+  ngOnInit(): void {
+    this.triggerInitialResize();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (!this.firstChange) {
+      if (changes.toggleCalculate && this.performanceForm.valid) {
+        this.checkCurveChanged();
```

Comment:
`if(changes.toggleCalculate && !changes.toggleCalculate.firstChange && ... )`

instead of having the this.firstChange variable